google-cloud-spanner/lib/google/cloud/spanner/instance.rb

```diff
@@ -188,8 +206,55 @@ def labels= labels
           )
         end
 
+        ##
+        # Update changes.
+        #  `display_name`, `labels`, `nodes`, `processing_units` can be
+        #  updated. `processing_units` and `nodes` can be used interchangeably
+        #  to update.
+        #
+        # @return [Instance::Job] The job representing the long-running,
+        #   asynchronous processing of an instance update operation.
+        # @raise [ArgumentError] if both processing_units or nodes are specified.
+        #
+        # @example
+        #   require "google/cloud/spanner"
+        #
+        #   spanner = Google::Cloud::Spanner.new
+        #
+        #   instance = spanner.instance "my-instance"
+        #   instance.display_name = "prod-instance"
+        #   instance.labels = { env: "prod", app: "api" }
+        #   instance.nodes = 2
+        #   # OR
+        #   # instance.processing_units = 500
+        #
+        #   job = instance.save
+        #
+        #   job.done? #=> false
+        #   job.reload! # API call
+        #   job.done? #=> true
+        #
+        #   if job.error?
+        #     status = job.error
+        #   else
+        #     instance = job.instance
+        #   end
+        #
         def save
-          job_grpc = service.update_instance @grpc
+          ensure_service!
+
+          field_mask = []
+          @orignal_values.each do |field, value|
+            field_mask << field unless @grpc[field.to_s] == value
+          end
+
+          if field_mask.include?(:node_count) &&
```

Comment:
Remove this check as well. Let the backend to return the error and keep a thin client.

Reply:
done